### PR TITLE
refactor: split ModifierDatabase from Modifiers

### PR DESCRIPTION
**Pull request by midgleyc**
Split "Modifiers" into "Modifiers" and "ModifierDatabase". I also shifted a bunch of functions around and tried to separate them into "regions" of similarity. Also, I moved all the fields to the top, and all the static initialisation together.

Roughly, ModifierDatabase tries to handle those actions which are not character-specific: reading from the database in modifiers.txt, writing to it, parsing it, parsing KoL descriptions.

There is one exception: `getNumericModifier` for FamiliarData sets the user's current familiar on the Modifiers object as a static field. There are a few of these fields. This is used in `Expression` when calculating familiar-specific modifiers. I feel like we shouldn't do it this way -- for example, we could instead pass in "optional overrides" to `eval` stored on the `Modifiers` object, non-statically, so they apply only while the given object is in use.

One function that was only used in `BasementRequest` was moved there. I felt based on how it worked it wasn't usable in general.

I also added a bunch of TODOs throughout the code that will hopefully be answered as I learn more, or with the benefit of more tests. For example, in `KoLCharacter` we call `resetModifiers` on ascension and on character reset, which resets the database of modifiers, and I think we may not need to do this (unless moving to / from TCRS). This is mostly important for tests, as `reset()` is called a lot.

### CODE REVIEW

**Comment by midgleyc** [comment]
I've gone back to initialising the ModifierDatabase in the Modifiers class due to the test failures. If anyone has an idea for a better way of doing this, please let me know.

**Review by jaadams5 — APPROVED** [review]
Perhaps longer than my attention span...

File: src/net/sourceforge/kolmafia/persistence/ModifierDatabase.java [hunk 1]
```diff
@@ -0,0 +1,1524 @@
...
+
+      newMods.setString(mod, value);
+    }
+
+    newMods.setString(StringModifier.MODIFIERS, string);
+
+    return newMods;
+  }
+
+  // TODO: what's the difference between these are the above?
```
Comment (Veracity0) [inline src/net/sourceforge/kolmafia/persistence/ModifierDatabase.java:662]: I believe that these are not in the blue "enchantments" section of a desc file. But by all means, check it out. 🙂

**Review by Veracity0 — APPROVED** [review]
What a big job. I like your organization.

Mostly. I like static Patterns that are used in a single method to be right in front of the method, since you easily see what it is while inspecting the method.

But your way is also a widely used code style, so, fine.

**Comment by Veracity0** [comment]
Wow. It got merged while I was performing MY code review.

Well, I approved it, but will you even see my comments now?

**Comment by midgleyc** [comment]
I created the PR and am watching it, so I see all the comments posted, whether the PR is open or not.

Thanks for the review :-D

**Comment by midgleyc** [comment]
> Mostly. I like static Patterns that are used in a single method to be right in front of the method, since you easily see what it is while inspecting the method.

I also like that style... when the file is short. If I can't keep it all in my head at once, I find it easier to have "fields", "constant fields", "methods" etc. sections.

For example, in the original Modifiers code there was a static initializer ~1000 lines in. I really don't like that!

**Pull request merged**